src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -0,0 +1,998 @@
+			int newSourceIndex = 0;
+			Map<Object, Object> params = new HashMap<>();
+			Map<Object, Object> changeAuxParams = new HashMap<>();
+			for (Entry<Integer, String> entry : listSources.entrySet()) {
+				if (entry.getValue().equals(sourceName)) {
+					newSourceIndex = entry.getKey();
+					break;
+				}
+			}
+			params.put(BarcoE2Constant.TYPE, BarcoE2Constant.DESTINATION_AUX_TYPE);
+			JsonNode response = getByMethod(BarcoE2Constant.METHOD_LIST_DESTINATIONS, params).get(BarcoE2ControllingMetric.AUX_DESTINATION.getName());
+			int currentAuxDestId = BarcoE2Constant.NOT_MATCH_AUX_ID;
+			for (int i = 0; i < response.size(); i++) {
+				JsonNode auxResponse = response.get(i);
+				if (auxResponse != null && Objects.equals(auxName, auxResponse.get(BarcoE2Constant.NAME).asText())) {
+					currentAuxDestId = response.get(i).get(BarcoE2Constant.ID).asInt();
+				}
+			}
+			changeAuxParams.put(BarcoE2Constant.ID, currentAuxDestId);
+			changeAuxParams.put(BarcoE2Constant.PGM_LAST_SRC_INDEX, newSourceIndex);
+			JsonNode changeAuxContentResult = getByMethod(BarcoE2Constant.METHOD_CHANGE_AUX_CONTENT, changeAuxParams);
+			if (changeAuxContentResult != null && changeAuxContentResult.get(BarcoE2Constant.SUCCESS_STATUS) != null) {
+				return changeAuxContentResult.get(BarcoE2Constant.SUCCESS_STATUS).asInt() == 0;
+			}
+			throw new CommandFailureException(this.getAddress(), "changeAuxContent", "Failed to call changeAuxContent");
+		} catch (Exception e) {
+			logger.error("Failed to assign source to super/aux destination");
+			throw new CommandFailureException(this.getAddress(), "changeAuxContent or changeSuperAuxContent", "Failed to assign source to this destination");
+		}
+	}
+
+	/**
+	 * Routing control: Get routing for super dest/aux
+	 *
+	 * @param isSuperDest boolean true/false if destination is screen/aux destination
+	 * @param stats Map of stats
+	 * @param controls List of AdvancedControllableProperty
+	 * @throws Exception Throw exception if failed to get json node,
+	 */
+	private void getSuperRoutingControl(boolean isSuperDest, Map<String, String> stats, List<AdvancedControllableProperty> controls) throws Exception {
+		String methodName = isSuperDest ? BarcoE2ControllingMetric.SUPER_SCREEN_DESTINATION.getName() : BarcoE2ControllingMetric.SUPER_AUX_DESTINATION.getName();
+		List<Integer> listDestIds = handleListSuperId(isSuperDest);
+		if (listDestIds.isEmpty()) {
+			stats.put(String.format(BarcoE2Constant.GROUP_DEST_HASH_TAG_DEST_NAME_COLON_MEMBER, methodName, BarcoE2Constant.NONE, BarcoE2Constant.NONE), BarcoE2Constant.NONE);
+			return;
+		}
+		List<String> sourceList = new ArrayList<>(getListSource().values());
+		// Super Screen/Aux dest loop
+		for (Integer listDestId : listDestIds) {
+			JsonNode response;
+			try {
+				response = getSuperRoutingControlJsonNode(isSuperDest, String.valueOf(listDestId));
+			} catch (Exception e) {
+				logger.error("Input from adapter properties does not match the device's response");
```

Comment:
effective error re-throwing, also original exception is getting lost (no 2nd parameters for ResourceNotReachableException). 
Is it necessary to catch exception here? Original stacktrace will be propagated anyways :)

Reply:
I deleted the logger.error here and add the 2nd parameter here. I think the exception should be catched and thrown here, because I want to make the error message more specific.